modules/file/file.go

```diff
@@ -1478,6 +1516,72 @@ func (r *run) checkHash(f fileEntry, hashtype checkType) {
 	return
 }
 
+func (r *run) checkBytes(f fileEntry) {
+	var (
+		err error
+	)
+	defer func() {
+		if e := recover(); e != nil {
+			err = fmt.Errorf("checkBytes() -> %v", e)
+		}
+	}()
+	reader := f.getReader()
+	defer f.Close()
+
+	// Must use the "label,search" for loop iteration because of
+	// map[string]search for parameters
+	for _, search := range r.Parameters.Searches {
+		if !search.isactive {
+			continue
+		}
+
+		for _, c := range search.checks {
+
+			decodedByte := make([]byte, hex.DecodedLen(len(c.bytes)))
+			_, err := hex.Decode(decodedByte, c.bytes)
+			if err != nil {
+				e := fmt.Sprintf("unable to decode byte query: %v ", err)
+				panic(e)
+			}
+			scansize := len(decodedByte) // Get the actual byte slice length
```

Comment:
@jvehent How would I also store the length when I decode the byte string?  Wouldn't it make more sense to determine the length of a specific check once, when it's being called?